Update friends controller to have create method

catherinemdean15 opened the pull request:
Add flash messages for successful and unsuccessful creation
Add search feature for friends
Add dashboard tests to list and add friends

(blank line)
Co-authored-by: Joe Jiang <69106714+ninesky00@users.noreply.github.com>
Co-authored-by: Cydnee Owens <67389821+cowens87@users.noreply.github.com>

catherinemdean15 committed:
Update friends controller to have create method
Add flash messages for successful and unsuccessful creation
Add search feature for friends
Add dashboard tests to list and add friends

(blank line)
Co-authored-by: Joe Jiang <69106714+ninesky00@users.noreply.github.com>
Co-authored-by: Cydnee Owens <67389821+cowens87@users.noreply.github.com>

ninesky00 commented on app/controllers/friends_controller.rb:
```diff
@@ -0,0 +1,19 @@
+  def friend_params
+    params.require(:friend).permit(:email)
+  end
+end
```
is the strong params necessary?

catherinemdean15 replied:
great question! I'm not totally sure. We can circle back!

ninesky00 commented on app/views/users/dashboard.html.erb:
```diff
@@ -4,10 +4,20 @@
-    <%= friend.email %>
-  <% end %>
+  <h3>Friends</h3>
+    <% if current_user.followings == [] %>
+      <%= "You currently have no friends" %>
```
This might touch on MVC principles

catherinemdean15 replied:
Great catch!

catherinemdean15 committed:
Update relationship tests for Friend and User
Create model method has_no_followers? in User
Create model test for has_no_followers?
Add before each section in User spec
Update dashboard view with has_no_followers? method
(blank line)
Co-authored-by: Joe Jiang <69106714+ninesky00@users.noreply.github.com>
Co-authored-by: Cydnee Owens <67389821+cowens87@users.noreply.github.com>
